packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1442,7 +1448,7 @@ class ComboBoxInternal extends React.Component<IComboBoxInternalProps, IComboBox
    * we do not have a valid index and we currently have a pending input value,
    * otherwise use the selected index
    * */
-  private _isOptionSelected(index: number | undefined): boolean {
+  private _isOptionHighlighted(index: number | undefined): boolean {
```

Comment:
Just to be clear, what (if anything) is different about the visible selection/hover behavior after all these ComboBox changes?

Reply:
I tried it and I'm not seeing any visual behavior differences (which is probably a good thing)

Reply:
Ah, yup! No visual behavior changes. The reason I separated it out was to make sure the visual logic was preserved while I made the changes I needed for the selection logic.